pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -210,6 +210,20 @@ func (m *AddOnManager) ApplyAddOn(addOn addon.AddOn, context *command.ExecutionC
 		return err
 	}
 
+	var depsNotInstalled []string
+	dependencies, err := addonMetadata.Dependency()
+	if err == nil && len(dependencies) > 0 {
+		for _, dep := range dependencies {
+			if m.IsInstalled(dep) {
+				continue
+			}
+			depsNotInstalled = append(depsNotInstalled, dep)
+		}
+		if len(depsNotInstalled) != 0 {
+			return errors.New(fmt.Sprintf("The add-on depends on: [%s]. Please install these first.", strings.Join(depsNotInstalled, ", ")))
+		}
```

Comment:
@anjannath This above logic should be in a function and the function can similar to `verifyRequiredAddons()`, else the code is getting bigger for `ApplyAddOn` which is not good programming practice.